File: data/EXP01/python/2564/2564.output_3.py
# @lc app=leetcode id=2564 lang=python3
class Solution:
    def substringXorQueries(self, s: str, queries: List[List[int]]) -> List[List[int]]:
        n = len(s)
        
        prefix_xor = [0] * n
        prefix_xor[0] = int(s[0])
        for i in range(1, n):
            prefix_xor[i] = prefix_xor[i-1] ^ int(s[i])
        
        ans = []
        for query in queries:
            first, second = query
            left, right = -1, -1
            min_len = float('inf')
            
            for i in range(n):
                if prefix_xor[i] ^ first == second:
                    left = i - min_len + 1
                    right = i
                    min_len = right - left + 1
            
            if left == -1:
                ans.append([-1, -1])
            else:
                ans.append([left, right])
        
        return ans

File: data/EXP01/golang_no_system_var/2564/2564.output_3.go
func substringXorQueries(s string, queries [][]int) [][]int {
	n := len(s)
	xorSum := make([]int, n+1)
	for i := 1; i <= n; i++ {
		xorSum[i] = xorSum[i-1] ^ int(s[i-1]-'0')
	}

	ans := make([][]int, len(queries))
	for i, query := range queries {
		left, right := query[0], query[1]
		xorValue := xorSum[right+1] ^ xorSum[left]

		leftIndex := -1
		for j := 0; j <= right; j++ {
			if xorSum[j] == xorValue {
				leftIndex = j
				break
			}
		}

		if leftIndex == -1 {
			ans[i] = []int{-1, -1}
		} else {
			rightIndex := leftIndex
			for j := right; j >= left; j-- {
				if xorSum[j] == xorValue {
					rightIndex = j
					break
				}
			}
			ans[i] = []int{leftIndex, rightIndex - 1} // Adjust the right endpoint by 1
		}
	}

	return ans
}
